File: cq-backend/src/main/java/br/com/cq/dominio/Revisao.java
package br.com.cq.dominio;

import java.util.List;

import org.joda.time.DateTime;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Builder;
import lombok.Data;

@Builder
@Data
@Document
public class Revisao {
	
	@Id
	private String id;
	private List<Questao> questoes;
	private Integer totalQuestoes;
	private Integer totalAcertos;
	private Integer totalErros;
	private Integer porcentagemAcertos;
	private Integer passo;
	private boolean concluido;
	private DateTime dataCriacao;

}

File: cq-frontend/src/components/dominio/Revisao.js
class Revisao {
    constructor (id, questoes, totalQuestoes, totalAcertos, totalErros, porcentagemAcertos,
                passo, concluido, dataCriacao) {
        this.id = id
        this.questoes = questoes
        this.totalQuestoes = totalQuestoes
        this.totalAcertos = totalAcertos
        this.totalErros = totalErros
        this.porcentagemAcertos = porcentagemAcertos
        this.passo = passo
        this.concluido = concluido
        this.dataCriacao = dataCriacao
    }
}
export default Revisao
